time_execution/backends/threaded.py

```diff
@@ -74,12 +75,17 @@ def send_metrics():
                 logger.warning('%r write failure %r', self.backend, exc)
 
         while self.has_work():
+            if self.thread:
```

Comment:
In the old daemon thread situation I believe the daemon thread was killed aburptly, without sending the remaining tasks. So nothing changed there.